tests/ui/consts/const-eval/ub-int-array.rs

```diff
@@ -10,54 +10,52 @@ union MaybeUninit<T: Copy> {
     init: T,
 }
 
+impl<T: Copy> MaybeUninit<T> {
+    const fn new(t: T) -> Self {
+        MaybeUninit { init: t }
+    }
+}
+
 const UNINIT_INT_0: [u32; 3] = unsafe {
-    [
-        MaybeUninit { uninit: () }.init,
-        //~^ ERROR evaluation of constant value failed
-        //~| uninitialized
-        1,
-        2,
-    ]
+    //~^ ERROR it is undefined behavior to use this value
+    //~| invalid value at [0]
+    mem::transmute([MaybeUninit { uninit: () }, MaybeUninit::new(1), MaybeUninit::new(2)])
 };
 const UNINIT_INT_1: [u32; 3] = unsafe {
-    mem::transmute(
-        [
-            0u8,
-            0u8,
-            0u8,
-            0u8,
-            1u8,
-            MaybeUninit { uninit: () }.init,
-            //~^ ERROR evaluation of constant value failed
-            //~| uninitialized
```

Comment:
At some point these got changed to longer test what they were actually meant to test: reporting the right index in the error message. So I changed the test to fix that again.